akka-remote/src/main/scala/akka/remote/artery/compress/OutboundActorRefCompression.scala

```diff
@@ -101,6 +107,39 @@ private[remote] class OutboundCompressionTable[T](system: ActorSystem, remoteAdd
 
 }
 
+class HashScanIndexMap[T](val size: Int) {
```

Comment:
the FastUtil is not an option, given the number of lines of code

`hashCode` for `ActorRef` is quick so the scan would be quick.
I like it